apollo-cache-interceptor/src/commonMain/kotlin/com/apollographql/apollo/interceptor/cache/ApolloCacheInterceptor.kt

```diff
@@ -17,42 +17,86 @@ import com.apollographql.apollo.interceptor.ApolloResponse
 import com.apollographql.apollo.api.parseData
 import com.apollographql.apollo.cache.normalized.internal.normalize
 import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
 import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.map
 
 @ApolloExperimental
 class ApolloCacheInterceptor<S>(private val store: S) : ApolloRequestInterceptor where S : WriteableStore, S : ReadableStore {
 
+  private fun <D : Operation.Data> proceed(request: ApolloRequest<D>, chain: ApolloInterceptorChain): Flow<ApolloResponse<D>> {
+    return chain.proceed(request).map {
+      if (it.response.data != null) {
+        writeToCache(request, it.response.data!!, chain.customScalarAdapters)
+      }
+      it.setFromCache(false)
```

Comment:
do we need this? I mean isn;t it set to false by default?

Reply:
Since ApolloResponse doesn't know anything about cache anymore, this data is in the execution context only known by the CacheInterceptor.